.github/workflows/pr-checks.yml

```
@@ -0,0 +1,464 @@
+    types: [opened, synchronize, reopened, ready_for_review]
+
+# Concurrency: Cancel previous runs when new changes are pushed
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+# Default permissions: read-only for all scopes
+permissions:
+  contents: read
+  packages: read
+  metadata: read
```

actionlint (1.7.7)

unknown permission scope "metadata". all available permission scopes are "actions", "attestations", "checks", "contents", "deployments", "discussions", "id-token", "issues", "packages", "pages", "pull-requests", "repository-projects", "security-events", "statuses"

(permissions)